website/addons/owncloud/utils.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from website.oauth.models import ExternalAccount
+from website.security import encrypt, decrypt
+
+
+class ExternalAccountConverter:
+    """
+        OwnCloudProvider from website.addons.owncloud.models hijacks the oauth
+        framework to serialize account credentials for owncloud.
+        ExternalAccountConverter provides a wrapper class for mapping between
+        the two.
+    """
+    def __init__(self, account=None, host=None, username=None, password=None):
```

Comment:
This seems like there should be three successful initialization scenarios:
1) an account is provided, but nothing else is
2) nothing is provided
3) a host, username, and password are provided, but account is not.

An error should be raised in any other case.

I'm also tempted to recommend breaking case 3 out into its own method -- `encrypt_external_account` or some such, so as to disambiguate the intended behavior when initializing this class, and allow this to accept only `account` or nothing. Creating a new `ExternalAccount` whenever you pass this raw credentials could be dangerous, as saving them (which I know isn't being done here) may result in EA's that violate uniqueness constraints.